### ISSUE TITLE
[Bug] Image filter is not applied in a specific case

### ISSUE THREAD

**Issue by Swati4star** [Swati4star/Images-to-PDF#218]
### Steps to reproduce
- Select 2-3 images
- Select `Filter Images`
- Apply some filter to image (blac&white or sepia)
- Select `Save current`
- Press back button
- Create PDF. Type some name.

### Expected behaviour
The first image should have the filter applied

### Actual behaviour
The filter is not applied to image.

**Comment by shiv07tiwari** [comment]
@Swati4star  In this, we are not pressing the "OK" button in Filter Images so as to reproduce the bug, right ?

**Comment by Swati4star** [comment]
Which `OK` button? There is no ok button in filters screen.

**Comment by shiv07tiwari** [comment]
Oh sorry, done button (top right)

**Comment by Swati4star** [comment]
Yes. exactly.

After pressing the save button, the filter should have been applied. But, it's not getting applied in case, we press back button

**Comment by shiv07tiwari** [comment]
So basically, we need to execute the code of "Done" button, when we press the back button also ?

**Comment by Swati4star** [comment]
Yes. 👍 
..and also when we press the back button on the top left corner.

**Comment by eddygta17** [comment]
@Swati4star 
>and also when we press the back button on the top left corner.

Shouldn't that be an option to cancel the applied filter?

![ezgif-1-d1b86bb1fa](https://user-images.githubusercontent.com/24777071/43236762-6d29c05c-90a4-11e8-8c29-d146962c6453.gif)

**Comment by Swati4star** [comment]
But, if the user has clicked save current filter button & press back, then the current filter should be applied. 

Also, in the gif, you are pressing back button after applying the filter on first image?

**Comment by eddygta17** [comment]
I am pressing back on keypad in the gif. 

Then how about replacing the arrow on the menubar with "CANCEL"

**Comment by Swati4star** [comment]
@eddygta17 Good idea! But, when user clicks on cancel, just show a confirmation dialog `Are you sure you want to remove filters from all your images?` 
If user says, ok, then go ahead. Else, do nothing.

**Comment by eddygta17** [comment]
Sure will do.
